pydantic/_internal/_generate_schema.py

```diff
@@ -1173,41 +1188,67 @@ def apply_model_validators(
 
 
 def apply_annotations(
-    schema: core_schema.CoreSchema, annotations: typing.Iterable[Any], definitions: dict[str, core_schema.CoreSchema]
-) -> core_schema.CoreSchema:
+    get_inner_schema: ModifyCoreSchemaWrapHandler,
+    annotations: typing.Iterable[Any],
+    definitions: dict[str, core_schema.CoreSchema],
+) -> ModifyCoreSchemaWrapHandler:
     """
     Apply arguments from `Annotated` or from `FieldInfo` to a schema.
     """
-    handler = CoreMetadataHandler(schema)
     for metadata in annotations:
-        schema = apply_single_annotation(schema, metadata, definitions)
+        if metadata is None:
+            continue
+        # check if this supports
```

Comment:
Was this comment supposed to say more?

Reply:
No I have no idea what where I was going with that; deleted it.